packages/dbml-core/src/parse/dbml/src/lib/lexer/lexer.ts

```diff
@@ -0,0 +1,567 @@
+    this.current += step;
+    this.start = this.current;
+  }
+
+  private closeGap() {
+    this.start = this.current;
+  }
+
+  private addToken(kind: SyntaxTokenKind) {
+    this.tokens.push(
+      SyntaxToken.create(
+        kind,
+        this.start,
+        this.current - this.start,
+        this.text.substring(this.start, this.current),
+      ),
+    );
+  }
+
+  lex(): Result<SyntaxToken[]> {
+    this.init();
+    while (!this.isAtEnd()) {
+      const c: string = this.peek()!;
+      switch (c) {
+        case ' ':
+          this.advance();
+          this.addToken(SyntaxTokenKind.SPACE);
+          break;
+        case '\r':
+          this.advance();
+          break;
+        case '\n':
+          this.advance();
+          this.addToken(SyntaxTokenKind.NEWLINE);
+          break;
+        case '\t':
+          this.advance();
+          this.addToken(SyntaxTokenKind.TAB);
+          break;
+        case ',':
+          this.advance();
+          this.addToken(SyntaxTokenKind.COMMA);
+          break;
+        case '.':
+          this.advance();
+          this.addToken(SyntaxTokenKind.DOT);
+          break;
+        case '/':
+          if (this.peek(1) === '/') {
+            this.singleLineComment();
+          } else if (this.peek(1) === '*') {
+            this.multilineComment();
+          } else {
+            this.advance();
+            this.addToken(SyntaxTokenKind.FORWARDSLASH);
+          }
+          break;
+        case '%':
+          this.advance();
+          this.addToken(SyntaxTokenKind.PERCENT);
+          break;
+        case '*':
+          this.advance();
+          this.addToken(SyntaxTokenKind.ASTERISK);
+          break;
+        case '+':
+          this.advance();
+          this.addToken(SyntaxTokenKind.CROSS);
+          break;
+        case '-':
+          this.advance();
+          this.addToken(SyntaxTokenKind.MINUS);
+          break;
+        case '(':
+          this.advance();
+          this.addToken(SyntaxTokenKind.LPAREN);
+          break;
+        case ')':
+          this.advance();
+          this.addToken(SyntaxTokenKind.RPAREN);
+          break;
+        case '[':
+          this.advance();
+          this.addToken(SyntaxTokenKind.LBRACKET);
+          break;
+        case ']':
+          this.advance();
+          this.addToken(SyntaxTokenKind.RBRACKET);
+          break;
+        case '{':
+          this.advance();
+          this.addToken(SyntaxTokenKind.LBRACE);
+          break;
+        case '}':
+          this.advance();
+          this.addToken(SyntaxTokenKind.RBRACE);
+          break;
+        case ';':
+          this.advance();
+          this.addToken(SyntaxTokenKind.SEMICOLON);
+          break;
+        case ':':
+          this.advance();
+          this.addToken(SyntaxTokenKind.COLON);
+          break;
+        case '<':
+          this.advance();
+          if (this.peek() !== '=') {
+            this.addToken(SyntaxTokenKind.LT);
```

Comment:
Instead of classifying for each operator `<=`, `>`, etc., should we use one kind `OPERATOR` for all of the operators? Doing that in the future if we want to support new operators, we don't need to parse again. Also, same operator in different contexts has different names/meaning (i.e. `>` this could be greater than in binary expression, but it could be many-to-one operator in relationship expression)